src/Util/DeepCopy.php

```diff
@@ -79,19 +88,47 @@ public function to(Model $destination)
      */
     public function with(array $references)
     {
-        $this->references = array_merge_recursive($this->references, $references);
+        $this->references = $references;
+
+        return $this;
+    }
+
+    public function excluding(array $exclusions)
+    {
+        $this->exclusions = $exclusions;
 
         return $this;
     }
 
+    /**
+     * Will extract non-numeric keys from the array.
+     *
+     * @param $array
+     *
+     * @return array
+     */
+    protected function extractKeys($array): array
+    {
+        $result = [];
+        foreach ($array as $key=>$val) {
+            if (is_numeric($key)) {
```

Comment:
shouldn't we use `is_int` here? For example, key `'12.345'` is numeric, but most likely not what you expect here.